File: core/src/main/java/no/elg/infiniteBootleg/args/Argument.java
package no.elg.infiniteBootleg.args;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * @author Elg
 */
@Target(ElementType.METHOD)
@Retention(RetentionPolicy.RUNTIME)
public @interface Argument {

    /**
     * @return Help string
     */
    String value();

    /**
     * @return Single char alias
     */
    char alt() default '\0';
}

File: core/src/main/java/no/elg/infiniteBootleg/args/Argument.kt
package no.elg.infiniteBootleg.args

/**
 * @author Elg
 */
@Target(AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.PROPERTY_SETTER)
@Retention(AnnotationRetention.RUNTIME)
annotation class Argument(
  /**
   * @return Help string
   */
  val value: String,
  /**
   * @return Single char alias
   */
  val alt: Char = '\u0000'
)
